### PR TITLE
issue #IQ-523 feat QUML compliance Hints and instruction changes

### PR DESCRIPTION
**Pull request by ramkumar-pacewisdom**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/IQ-523" title="IQ-523" target="_blank">Minor-EnhancementIQ-523</a>  Quml Compliance changes for Hints and Instructions in Collection editor.
  </td></table>
  <br />
 

In this PR QUML Compliance for Hint and Instructions are added.

### CODE REVIEW

**Comment by rajnishdargan** [comment]
@ramkumar-pacewisdom There is one code smell in the PR, please fix it.

File: projects/questionset-editor-library/src/lib/components/question/question.component.ts [hunk 1]
```diff
@@ -899,7 +914,7 @@ export class QuestionComponent implements OnInit, AfterViewInit, OnDestroy {
   getDefaultSessionContext() {
     return _.omitBy(_.merge(
       {
-        creator: _.get(this.editorService.editorConfig, 'context.user.fullName'),
+        // creator: _.get(this.editorService.editorConfig, 'context.user.fullName'),
```
[the remark this answers is not in the document]
Comment (ramkumar-pacewisdom) [reply projects/questionset-editor-library/src/lib/components/question/question.component.ts:917]: sure.

File: projects/questionset-editor-library/src/lib/components/question/question.component.spec.data.ts [hunk 2]
```diff
@@ -151,22 +152,22 @@ export const mockData = {
               {
                 label: "<p>option 1</p>",
                 value: 0,
-                hints: {
-                  en: "test hint 1",
+                hint: {
```
Comment (rajnishdargan) [inline projects/questionset-editor-library/src/lib/components/question/question.component.spec.data.ts:155]: This is not as per QuML spec, please check and update the same. Here is the example for storing hints in interaction- 
`hint: "70c82bf5-9459-4c43-8897-0e58b7e1da64"`

File: projects/questionset-editor-library/src/lib/components/question/question.component.spec.data.ts [hunk 3]
```diff
@@ -151,22 +152,22 @@ export const mockData = {
...
+                  '70c82bf5-9459-4c43-8897-0e58b7e1da62':'test hint 1'
                 },
               },
               {
                 label: "<p>option 2</p>",
                 value: 1,
-                hints: {
-                  en: "test hint 2",
+                hint: {
```
Comment (rajnishdargan) [inline projects/questionset-editor-library/src/lib/components/question/question.component.spec.data.ts:162]: @ramkumar-pacewisdom Please check the above comment

File: projects/questionset-editor-library/src/lib/services/editor/editor.service.spec.data.ts [hunk 4]
```diff
@@ -31,15 +31,15 @@ export const rootNode = {
                     contentDisposition: 'inline',
                     lastUpdatedOn: '2022-04-05T06:40:11.752+0000',
                     contentEncoding: 'gzip',
-                    showSolutions: false,
+                    showSolutions: 'No',
```
Comment (rajnishdargan) [inline projects/questionset-editor-library/src/lib/services/editor/editor.service.spec.data.ts:34]: @ramkumar-pacewisdom This is not as per new format please check the below document for the same - https://project-sunbird.atlassian.net/wiki/spaces/QB/pages/3299803150/Proposed+QuML+Spec+Changes?focusedCommentId=3316875648

File: projects/questionset-editor-library/src/lib/services/editor/editor.service.spec.data.ts [hunk 5]
```diff
@@ -275,6 +268,7 @@ export const questionsData = [
         channel: '01309282781705830427',
         responseDeclaration: {
             response1: {
+                maxScore: 1,
```
Comment (rajnishdargan) [inline projects/questionset-editor-library/src/lib/services/editor/editor.service.spec.data.ts:271]: @ramkumar-pacewisdom As per the new format we cannot store the maxScore property inside the responseDeclaration. Please check this document https://project-sunbird.atlassian.net/wiki/spaces/QB/pages/3317628929/QuML+compliance+implementation+changes+for+editor+and+player#maxScore

**Review by rajnishdargan — CHANGES_REQUESTED** [review]
@ramkumar-pacewisdom Please resolve the mentioned comments.

File: projects/questionset-editor-library/src/lib/services/editor/editor.service.spec.data.ts (hunk 4 above)
Comment (ramkumar-pacewisdom) [reply projects/questionset-editor-library/src/lib/services/editor/editor.service.spec.data.ts:34]: updated

File: projects/questionset-editor-library/src/lib/components/question/question.component.spec.data.ts (hunk 3 above)
Comment (ramkumar-pacewisdom) [reply projects/questionset-editor-library/src/lib/components/question/question.component.spec.data.ts:162]: updated

**Review by rajnishdargan — APPROVED** [review]

**Review by sajeshkayyath — APPROVED** [review]

**Pull request merged**